src/components/Link.tsx

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+
+type LinkVariantProps = React.HTMLProps<HTMLAnchorElement> & {
+  variant?: "link";
+  href: string;
+};
+type NavVariantProps = React.HTMLProps<HTMLLIElement> & {
+  isActive: boolean;
+  variant: "nav";
+  onClick: () => void;
+};
+type Props = LinkVariantProps | NavVariantProps;
+
+/**
+ * A reusable link component with animated underline when the link is active.
+ *
+ * Has two variants:
+ *  1. "link" (default): renders an <a> tag with hover/focus styling.
+ *  2. "nav": renders a <button> with an onClick callback. Active state styling
+ *     is handled by an isActive prop, rather than directly by hover/focus.
+ */
+const Link = (props: Props) => {
+  const { variant, children } = props;
+
+  const getActiveStyle = (): string => {
+    if (variant === "nav") {
+      return props.isActive
+        ? "after:right-0 after:opacity-1"
+        : "after:right-full after:opacity-0";
+    } else {
+      return `
+        after:right-full
+        hover:after:right-0
+        focus:after:right-0
+
+        after:opacity-0
+        hover:after:opacity-100
+        focus:after:opacity-100
+      `;
+    }
+  };
+
+  const className = `
+    cursor-pointer
+    relative 
+ 
+    after:block
+    after:absolute
+    after:border
+    after:h-px
+    after:top-full
+    after:left-0
+    after:transition-all
+    after:duration-300
+
+    ${getActiveStyle()}
+  `.trim();
```

Comment:
I was a bit confused momentarily, but it appears there is a bug due to how whitespaces are handled here. When viewing the page, focus doesn't work properly on the "WoW made easy" button because the classes have turned into this:

![image](https://user-images.githubusercontent.com/1728144/228054822-9a19b38f-791f-4a79-8cbc-229c965a40c9.png)

Reply:
Ah right, thanks for spotting. I think that final `hover:opacity-50` was an oversight on my part and doesn't need to be there at all, so will fix this.

Reply:
I can confirm it works now. 😄